tracing-core/src/field.rs

```diff
@@ -922,7 +930,18 @@ impl FieldSet {
     {
         ValueSet {
             fields: self,
-            values: values.borrow(),
+            values: Values::Mixed(values.borrow()),
+        }
+    }
+
+    /// Returns a new `ValueSet` for `values`. These values must exactly
+    /// correspond to the fields in this `FieldSet`.
```

Comment:
```suggestion
    /// correspond to the fields in this `FieldSet`.
    ///
    /// # Safety Invariant
    /// The length of `values` must exactly match `self.len()`. This is validated
    /// in debug builds but not in release builds for performance. Violating this
    /// invariant may lead to panics or incorrect behavior.
```
This is already documented at https://github.com/tokio-rs/tracing/pull/3398/files#diff-7513a604e72baf1a41f089a09846b836d8abd862289efed2f36dfb90b4571d39R176 but I'd find such clarification here useful!

Reply:
I'll add a small clarification, but I won't call it a "safety invariant"--violating this precondition won't cause memory safety issues (thus this method is not `unsafe`).

Reply:
I think that's fine.